### DISCUSSION TITLE
What Version to Use

### DISCUSSION THREAD

**Discussion by SimplyFriday** [excaliburjs/Excalibur#1995] [Q&A] [answered]
Hello,

I've written one game for a jam a while back with Excalibur 24.5. I ran into some odd issues, but was able to work around them. Somebody mentioned that a lot of the issues I was having are resolved in the current build.

Is the "next" branch (0.25.0-alpha.xxx) stable enough to start using, or should I stick with 0.24.5 until 0.25 is released?

**Comment by eonarheim** [comment, marked as answer]
@josh-greenlaw All the big stuff is merged to main for 0.25.0, some of the documentation is lagging (specifically the collision system updates). There are a few breaking changes in 0.25.0, which I'll be writing up a guide. The changelog is a good summary of the changes you can expect.

If you are okay with some breaking changes from 0.24 and some polishing/bug-fixing in the latest 0.25 versions, it would be awesome to get feedback on it from you either on needed documentation or examples. 

My plan given my current list is to cut 0.25.0 in the next month September-ish (polish/bugs willing)

Otherwise if you don't want to experiment just yet, I'd stay with 0.24.5 👍

**Comment by eonarheim** [comment]
@josh-greenlaw  Also I hope the game jam went well! I'd be interested in a list of the issues to be sure I've got them tracked/addressed/documented

**Reply by SimplyFriday** [reply]
Sounds good, I'll start using the 0.25 alpha and provide any feedback. The jam went well, considering it was my first game. I'm looking forward to the next game new, hoping to make something with multiplayer, ideally client/server! 

The main issue I was having was with z index not working properly on Actors added later during a Scene's lifetime. Something about the b-tree logic being wonky, but I believe that was all simplified. 

Another issue that I had frequently was that whenever I added a `Timer` to an `Actor` and called one of the Actor's methods from within the Timer, `this` would be a `Timer` type, not whatever `Actor` the method resided in. I resolved it be extending `Timer` and giving it a parentActor property. Then in the Actor's method I would get the parent actor via `let actor = (this as undefined as TimerExtensionClass).parentActor;` I suspect this is a limitation of TypeScript/JavaScript though, not Excalibur.

**Reply by kamranayub** [reply]
Thanks, we'll capture this in an issue!
